pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```diff
@@ -0,0 +1,218 @@
+
+        final String truststorePath =
+                cmd.getOptionValue("truststore") == null ? Constants.DEFAULT_TRUSTSTORE_PATH : cmd.getOptionValue("truststore");
+        final boolean validateHostname = cmd.getOptionValue("validatehost") == null ? false : true;
+
+        final String username = cmd.getOptionValue("username") == null ? Constants.DEFAULT_USERNAME : cmd.getOptionValue("username");
+        final String password = cmd.getOptionValue("password") == null ? Constants.DEFAULT_PASSWORD : cmd.getOptionValue("password");
+
+        SecureBatchReader reader = new SecureBatchReader(scope, stream, controllerURI,
+                truststorePath, validateHostname,
+                username, password);
+        reader.read();
+    }
+
+    private static Options getOptions() {
+        final Options options = new Options();
+        options.addOption("s", "scope", true, "The scope name of the stream to read from.");
+        options.addOption("n", "stream", true, "The name of the stream to read from.");
+        options.addOption("u", "uri", true, "The URI to the controller in the form tls://host:port");
+        options.addOption("t", "truststore", true,
+                "The location of .pem truststore file in the file system to use by this application process.");
+        options.addOption("v", "validatehost", false, "Whether to verify server's hostname.");
+        options.addOption("a", "username", true,
+                "The account username to use by the client for authenticating to the server.");
+        options.addOption("p", "password", true,
+                "The account password to use by the client for authenticating to the server.");
+        return options;
+    }
+
+    private static CommandLine parseCommandLineArgs(Options options, String[] args) throws ParseException {
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = parser.parse(options, args);
+        return cmd;
+    }
+
+    private int readFromSegments(BatchClientFactory batchClient, List<SegmentRange> segments) {
+        List<Integer> batchEventCountList = new ArrayList<>();
+        JavaSerializer<String> serializer = new JavaSerializer<>();
+        batchEventCountList = segments
+                .stream()
+                .map(segment -> {
+                    SegmentIterator<String> segmentIterator = batchClient.readSegment(segment, serializer);
+                    int numEvents = 0;
+                    try {
+                        String id = String.valueOf(Thread.currentThread().getId());
+                        while (segmentIterator.hasNext()) {
+                            String event = segmentIterator.next();
+                            System.out.println("Done reading event by thread " + id + ": " + event);
+                            numEvents++;
+                        }
+                    } catch (Exception e) {
+                        e.printStackTrace();
```

Comment:
So if you get an error you will return a partial result. This is incorrect. Bubble up the exception.

Reply:
Yes Andrei

Reply:
This is not done.

Reply:
Yes Andrei.